src/data/mv_polynomial/equiv.lean

```diff
@@ -319,6 +321,272 @@ begin
   simp only [mv_polynomial.fin_succ_equiv_apply, mv_polynomial.eval₂_hom_C],
 end
 
+lemma fin_succ_equiv_add {n : ℕ} {R : Type u} [comm_semiring R]
+  (f g : mv_polynomial (fin (n + 1)) R) : fin_succ_equiv R n (f + g)
+  = fin_succ_equiv R n f + fin_succ_equiv R n g := by simp
+
+lemma fin_succ_equiv_mul {n : ℕ} {R : Type u} [comm_semiring R]
+  (f g : mv_polynomial (fin (n + 1)) R) :
+  fin_succ_equiv R n (f * g) = fin_succ_equiv R n f * fin_succ_equiv R n g := by simp
+
+lemma fin_succ_equiv_zero {n : ℕ} {R : Type u} [comm_semiring R] :
+  fin_succ_equiv R n 0 = 0 := by simp
```

Comment:
These lemmas shouldn't exist, we already have them as `fin_succ_equiv.map_add` etc. When you add a new lemma that can be solved by `simp`, you should first try `squeeze_simp?` to see if a single lemma is solving it.

Reply:
`squeeze_simp?` in the proof of `fin_succ_equiv_add` is giving me the following error:

```
invalid simplification lemma 'alg_equiv.of_linear_equiv_symm' (use command 'set_option trace.simp_lemmas true' for more details)
```
I am doing something wrong? Using `squeeze_simp` gives `simp only [fin_succ_equiv_apply, coe_eval₂_hom, eval₂_add]`. 

I guessed that I should use `fin_succ_equiv.map_mul` and `fin_succ_equiv.map_zero`. However I couldn't use `squeeze_simp?` to avoid having `fin_succ_equiv_X_eq_zero` and `fin_succ_equiv_X_ne_zero`.